types/README.md

```diff
@@ -6,3 +6,41 @@
  * [`unique_ptr`](unique_ptr): `std::unique_ptr` with different element types
  * [`variant`](variant): `std::variant` with `Switch` column type
  * [`vector`](vector): `std::vector` with all `[Split]Index{32,64}` column types
+
+## Nesting of Types
+
+In the RNTuple Validation Suite, we cannot test all possible nesting combinations of supported types.
+Furthermore, we assume that the column encoding on disk is orthogonal to the nesting of fields.
+We therefore cover a strategic selection based on field structural roles that is described below.
+Implementations are required to exercise specific details in their own unit tests.
```

Comment:
I guess we cannot "require" implementations to do their tests; maybe should be phrased as "should"